app/decorators/staff/time_slot_decorator.rb

```diff
@@ -81,31 +77,23 @@ def linked_title
   end
 
   def display_presenter
-    if object.program_session.present?
-      speaker_names(object.program_session)
-    else
-      object.presenter
-    end
+    speaker_names(object.program_session) || object.presenter
```

Comment:
Yeah, I could optimize the check a bit by doing
```
program_session && speaker_names(...) || object.presenter
```
That way, it'd short-circuit immediately if there's no program_session. That's about as complex an expression as I would use short-circuiting for. #try does bring a performance penalty since it has to do a couple method calls internally.